src/main/java/crawlercommons/sitemaps/sax/DelegatorHandler.java

```diff
@@ -70,30 +72,33 @@ public void startElement(String uri, String localName, String qName, Attributes
         if (elementStack.isEmpty() || delegate == null) {
             startRootElement(uri, localName, qName, attributes);
         } else {
-            elementStack.push(qName);
+            elementStack.push(localName);
         }
         if (delegate != null) {
             delegate.startElement(uri, localName, qName, attributes);
         }
     }
 
     private void startRootElement(String uri, String localName, String qName, Attributes attributes) {
-        elementStack.push(qName);
-        if ("sitemapindex".equals(qName)) {
-            delegate = new XMLIndexHandler(url, elementStack, strict);
-        } else if ("urlset".equals(qName)) {
-            delegate = new XMLHandler(url, elementStack, strict);
-        } else if ("feed".equals(qName)) {
+        elementStack.push(localName);
+        
+        if ( Namespace.SITEMAP.equals(uri) ) {
```

Comment:
same here - why don't we consider only localnames regardless of the NS? What if the NS is not set?